ui/app/contexts/metametrics.js

```diff
@@ -55,45 +52,65 @@ export function MetaMetricsProvider ({ children }) {
     return unlisten
   }, [history])
 
+  /**
+   * track a metametrics event
+   *
+   * @param {import('../../../shared/modules/metametrics').MetaMetricsEventPayload} - payload for event
+   * @returns undefined
+   */
+  const trackEvent = useMemo(() => {
+    const referrer = confirmTransactionOrigin ? { url: confirmTransactionOrigin } : undefined
+    const page = {
+      path: currentPath,
+    }
+    return getTrackMetaMetricsEvent(global.platform.getVersion(), () => ({
+      context: {
+        referrer,
+        page,
+      },
+      environmentType,
+      network,
+      participateInMetaMetrics,
+      metaMetricsId,
+      metaMetricsSendCount,
```

Comment:
I noticed that `chainId` and `locale` are missing here. Those are probably worth including.